rfd/0001-general-design.md

```diff
@@ -0,0 +1,433 @@
+---
+authors: Anton Repin (github.com/xdire)
+state: wip
+---
+
+# RFD 1 - General Design
+
+## What
+
+This RFD Details the aspects of implementation of TCP load balancer
+up the following minimal requirements:
+
+### Library
+- A least connections forwarder implementation that tracks the number of connections per upstream.
+- A per-client connection rate limiter implementation that tracks the number of client connections.
```

Comment:
This will be addressed in the design update:

The current path is:
- As mentioned in mTLS section we decide the connection to be accepted/dropped by the decisions of certificate validity and CN=AccessKey for the Frontend record which designates the customer pool
- As there can be multiple Frontends working on the same server, we can implement TokenBucket algorithm for each Frontend UUID to have it's own TokenBucket updated on the next connection through designated port (ports not in the scope of this design)